frontend/js/src/user/UserFeedback.tsx

```diff
@@ -175,25 +175,23 @@ export default class UserFeedback extends React.Component<
       recordingMbidFeedbackMap,
     } = this.state;
 
-    let recording_msids = "";
-    let recording_mbids = "";
+    let recording_msids: string[] = [];
+    let recording_mbids: string[] = [];
     if (feedback?.length && currentUser?.name) {
       recording_msids = feedback
-        .map((item) => item.recording_msid)
+        .map((item) => item.recording_msid!)
```

Comment:
I'm not a big fan of those typescript `!` to resolve type issues, they tend to hide the problem rather than fix it explicitly.
I do understand that in this case it's typescript not understanding that our map + filter operation results in an array of strings (and not string |undefined | null), but I suggest this more declarative approach:

```typescript
    if (feedback?.length && currentUser?.name) {
      const recording_msids = feedback
        .map((item) => item.recording_msid)
        // Only request non-undefined and non-empty string
        .filter(Boolean)
        // Only request feedback we don't already have
        .filter((msid) => !has(recordingMsidFeedbackMap, msid));
      const recording_mbids = feedback
        .map((item) => item.recording_mbid)
        // Only request non-undefined and non-empty string
        .filter(Boolean)
        // Only request feedback we don't already have
        // @ts-ignore
        .filter((mbid) => !has(recordingMbidFeedbackMap, mbid)) as string[];
        […]
`
``

Note that the MSID array doesn't need anything done to it and already has the expected type, and we add the simple `as string[];` at the end of the MBID array declaration.

I think it's a bit easier to read and future developers have less chance of missing the `!`.

Reply:
Oh yes! Learnt something new. Changing these